src/k8s-extension/azext_k8s_extension/partner_extensions/ContainerInsights.py

```diff
@@ -624,7 +624,9 @@ def _ensure_container_insights_dcr_for_monitoring(cmd, subscription_id, cluster_
     except HttpResponseError as ex:
         raise ex
 
-    dataCollectionRuleName = f"MSCI-{cluster_name}-{cluster_region}"
+    dataCollectionRuleName = f"MSCI-{workspace_region}-{cluster_name}"
+    # Max length of the DCR name is 64 chars
+    dataCollectionRuleName = dataCollectionRuleName[0:64]

```

Comment:
Will it impact existing DCRs that may be longer than 64 chars?

Reply:
No.  There are no existing customers who are using longer than 64 chars. This we found during internal testing.